clean/lafayette_pd.py

```diff
@@ -419,13 +422,348 @@ def clean_cprr():
+    df.loc[:, 'investigator_rank'] = parts[0].fillna('')
+    df.loc[:, 'investigator_first_name'] = parts[1].fillna('')\
+        .str.replace(r'\.', '', regex=True)
+    df.loc[:, 'investigator_last_name'] = parts[2].fillna('')
+    return df.drop(columns=['investigator', 'assigned_investigator'])
+
+
+def extract_action_from_disposition_14(df):
+    df.loc[:, 'action'] = df.disposition.str.lower().str.strip().fillna('')\
+        .str.replace(r'^sus?tained ?\/ ?(.+) ?', r'\1', regex=True)\
+        .str.replace(r'\.', '', regex=True)\
+        .str.replace(r'lor', 'letter of reprimand', regex=False)\
+        .str.replace(r'(not sustained|unfounded|withdrawn|unfounded on all officers|'
+                     r'\bsust\b|complaint|exonorated|sent to hr)', '', regex=True)\
+        .str.replace(r'(officer resigned ?(prior)? ?(to)? ?(termination)?|resigned sts|'
+                     r'resigned under invest)', 'resigned', regex=True)\
+        .str.replace(r'(all focus officers listed received counseling forms|'
+                     r'counseli ?ng form)|loc', 'letter of counseling', regex=True)\
+        .str.replace('(retired under investigation)', 'retired', regex=True)\
+        .str.replace(r' ?(\d+) days? ?(suspension)? ?', r'\1-day suspension ', regex=True)\
+        .str.replace(r'excessive force/ failure to complete report/sustained (2days)|^2days$',
+                     '2-day suspension', regex=True)
+    actions = df.action.str.extract(r'((.+)suspension(.+)|(.+)terminated(.+)|(.+)resigned(.+)|'
+                                    r'(.+)letter of counseling(.+)|(.+)letter of reprimand(.+))')
+
+    df.loc[:, 'action'] = actions[0].fillna('')\
+        .str.replace(r'(.+)letter of counseling(.+)', 'letter of counseling', regex=True)\
+        .str.replace(r'(\w+)3', '3', regex=True)\
+        .str.replace(r' \((\w+)\)$', '', regex=True)\
+        .str.replace('terminated  resigned/', 'terminated; resigned', regex=False)
+    return df
+
+
+disposition_14_lookup = [
+    ['sustained; resigned', 'sustained/ nboc/resigned',
+     'sustained/resigned prior to termination', 'sustained/resigned',
+     'sustained/ resigned under invest.', 'terminated sust. resigned/', 'resigned sts.'],
+    ['unfounded', '/loc/sta rling- unfounded- crozier'],
+    ['exonerated', 'exonorated'],
+    ['justified', 'justified use of force', 'justified use of', 'justified/use of force'],
+    ['not sustained; sustained', 'not sustained/anboc- sustained deficiency'],
+    ['not sustained', '; not sustained- deroussel',
+     'not sustained/ds- counseling form (ryan beard) unprofessional conduct'],
+    ['not sustained; resigned', 'not sustained/ resigned'],
+    ['sustained', 'sustained/ lor', 'sustained/loc', 'sustained/ training',
+     'ned fowler- sustained- cubo/ letter of repremaind gabe thompson- sustained- cubo/ counseling form'],
+    ['withdrawn'],
+    ['no violation', 'no violation et', 'noviolation/close'],
+    ['resigned', '/ resigned'],
```

Comment:
Unless the leading and trailing slashes and spaces have meaning, I suggest stripping them from all strings. I think it makes the lookup table be more useful against more variations.